### ISSUE TITLE
ZAP retrieve list of alerts

### ISSUE THREAD

**Issue by JordanGS** [zaproxy/zaproxy#2762]
Currently there is a bug where the exported XML and HTML reports are not 1 to 1 with what is shown in the GUI, for the last two days i have been trying various methods, but mostly trying to get the following call to work:

[1]

```
List<Alert>allAlerts =newArrayList<>();
TableAlert tableAlert =getModel().getDb().getTableAlert();
```

for either of the following two packages

[2]

```
org.parosproxy.paros.extension.report
or
org.zaproxy.zap.extension.exportreport
```

I am willing to fix up the original reports and make them compatible with the above call so that they no longer use `siteXML` method which does not return the proper count. But i cannot seem to get the above [1] to return any results, is there any other way to get the same list of alerts that the GUI uses or how would i get the above call to return results from an extension?

Thank you very much!

**Comment by JordanGS** [comment]
OWASP ZAP Developer Group thread:
https://groups.google.com/d/topic/zaproxy-develop/K4-Fc3FytjY/discussion

**Comment by kingthorin** [comment]
@JordanGS  is thus still an issue?

**Comment by kingthorin** [comment]
@JordanGS is this still an issue?

**Comment by JordanGS** [comment]
@kingthorin Not sure, leave it open for now. I have asked for access to my old test sites and will be able to evaluate once i have that. Not sure if i'll get access this week or in 3-4 weeks. I just got approved for VPN.

**Comment by kingthorin** [comment]
@JordanGS any news?

**Comment by JordanGS** [comment]
Will test tomorrow and update thread accordingly.

Cheers, Jordan.

**Comment by JordanGS** [comment]
I haven't been able to replicate the issue with a new session. Not sure if that's because I only have bodgeit available for testing at the moment or because it's been fixed.

I'm trying to find my original sessions which had the issue and that's proving to be troublesome. I think I still have a copy at my old workplace but I won't be able to go there until Thursday/Friday.

**Comment by JordanGS** [comment]
@kingthorin i don't have my original session and i currently don't have access to a site where i could replicate the issue. It was never an issue with bodgeit store. As such i can't verify if the issue was resolved or not.

I've requested access to a test site from a source where i discovered the issue but i have no eta on how long it will take to get access. Perhaps mark it as resolved but also flag it as a come back to it later?
